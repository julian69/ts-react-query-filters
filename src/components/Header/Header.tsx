import { FC } from 'react';
import { Link } from 'react-router-dom';

import logUrl from '@/assets/logo.svg';

const Header: FC = () => (
  <header className="flex justify-center pt-6">
    <Link to="https://tanstack.com/" target="_blank">
      <img src={logUrl} alt="hero" width="50" />
    </Link>
  </header >
);

export default Header;
