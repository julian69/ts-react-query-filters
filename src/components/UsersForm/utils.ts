import { IFilters } from "@/types/users";

export const deserializeUrl = (params: URLSearchParams): IFilters => ({
  name: params.get('name') || '',
  email: params.get('email') || '',
  company: params.get('company') || '',
  zipcode: params.get('zipcode') || '',
});


