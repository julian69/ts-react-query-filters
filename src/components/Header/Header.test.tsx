import '@testing-library/jest-dom';
import { render } from '@/test/utils';
import { describe, expect, it } from 'vitest';

import Header from './Header';

describe('<Header />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});