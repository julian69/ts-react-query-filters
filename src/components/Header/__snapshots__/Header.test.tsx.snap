// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > should render correctly 1`] = `
<DocumentFragment>
  <header
    class="flex justify-center pt-6"
  >
    <a
      href="https://tanstack.com/"
      target="_blank"
    >
      <img
        alt="hero"
        src="/src/assets/logo.svg"
        width="50"
      />
    </a>
  </header>
</DocumentFragment>
`;
